src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/cascaded_document_layout.tsx

```diff
@@ -0,0 +1,226 @@
+import type { DataTableRecord } from '@kbn/discover-utils';
+import { useDiscoverServices } from '../../../../../hooks/use_discover_services';
+import { useScopedServices } from '../../../../../components/scoped_services_provider/scoped_services_provider';
+import { useCurrentTabSelector, useAppStateSelector } from '../../../state_management/redux';
+import {
+  useEsqlDataCascadeRowHeaderComponents,
+  useEsqlDataCascadeHeaderComponent,
+  ESQLDataCascadeLeafCell,
+  type ESQLDataGroupNode,
+} from './blocks';
+import { cascadedDocumentsStyles } from './cascaded_documents.styles';
+import { type CascadedDocumentsRestorableState } from './cascaded_documents_restorable_state';
+import { useEsqlDataCascadeRowActionHelpers } from './blocks/use_row_header_components';
+import {
+  useDataCascadeRowExpansionHandlers,
+  useGroupedCascadeData,
+  useScopedESQLQueryFetchClient,
+} from './hooks';
+import type { DiscoverStateContainer } from '../../../state_management/discover_state';
+
+export { getESQLStatsQueryMeta };
+
+export interface ESQLDataCascadeProps extends Omit<UnifiedDataTableProps, 'ref'> {
+  defaultFilters?: Filter[];
+  viewModeToggle: React.ReactElement | undefined;
+  cascadeGroupingChangeHandler: (cascadeGrouping: string[]) => void;
+  cascadeConfig: CascadedDocumentsRestorableState;
+  togglePopover: ReturnType<typeof useEsqlDataCascadeRowActionHelpers>['togglePopover'];
+  queryMeta: ESQLStatsQueryMeta;
+  registerCascadeRequestsInspectorAdapter: (requestAdapter: RequestAdapter) => void;
+}
+
+const ESQLDataCascade = React.memo(
+  ({
+    rows,
+    dataView,
+    viewModeToggle,
+    cascadeConfig,
+    cascadeGroupingChangeHandler,
+    togglePopover,
+    queryMeta,
+    registerCascadeRequestsInspectorAdapter,
+    ...props
+  }: ESQLDataCascadeProps) => {
+    const [query, defaultFilters] = useAppStateSelector((state) => [state.query, state.filters]);
+    const [globalState, esqlVariables] = useCurrentTabSelector((state) => [
+      state.globalState,
+      state.esqlVariables,
+    ]);
+    const globalFilters = globalState?.filters;
+    const globalTimeRange = globalState?.timeRange;
+    const { scopedProfilesManager } = useScopedServices();
+    const { expressions } = useDiscoverServices();
+
+    const cascadeRequestsInspectorAdapter = useRef<RequestAdapter>(new RequestAdapter());
+
+    useEffect(() => {
+      registerCascadeRequestsInspectorAdapter(cascadeRequestsInspectorAdapter.current);
+    }, [registerCascadeRequestsInspectorAdapter]);
+
+    const cascadeGroupData = useGroupedCascadeData({
+      cascadeConfig,
+      rows,
+      queryMeta,
+      esqlVariables,
+    });
+
+    const fetchCascadeData = useScopedESQLQueryFetchClient({
+      query: query as AggregateQuery,
+      dataView,
+      data: props.services.data,
+      esqlVariables,
+      expressions,
+      filters: [
+        ...(globalFilters?.filter((f) => f.meta.disabled === false) ?? []),
+        ...(defaultFilters ?? []),
+      ],
```

Comment:
What is the use case for adding DSL filters? Currently, there are no flows on Discover page where DSL filters would need to be applied to an ES|QL query.

Not a blocker for the merge.